NorthstarDLL/mods/modsavefiles.cpp

```diff
@@ -0,0 +1,483 @@
+		g_pSquirrel<context>->pushinteger(sqvm, (int)(fs::file_size(dir / fileName) / 1024));
+	}
+	catch (std::filesystem::filesystem_error const& ex)
+	{
+		spdlog::error("GET FILE SIZE FAILED! Is the path valid?");
+		g_pSquirrel<context>->raiseerror(sqvm, ex.what());
+		return SQRESULT_ERROR;
+	}
+	return SQRESULT_NOTNULL;
+}
+
+// void NSDeleteFile(string file)
+ADD_SQFUNC("void", NSDeleteFile, "string file", "", ScriptContext::SERVER | ScriptContext::CLIENT | ScriptContext::UI)
+{
+	Mod* mod = g_pSquirrel<context>->getcallingmod(sqvm);
+
+	fs::path dir = savePath / fs::path(mod->m_ModDirectory).filename();
+	std::string fileName = g_pSquirrel<context>->getstring(sqvm, 1);
+	if (!IsPathSafe(fileName, dir))
+	{
+		g_pSquirrel<context>->raiseerror(
+			sqvm,
+			fmt::format("File name invalid ({})! Make sure it results in a path inside your mod's save folder.", fileName, mod->Name)
+				.c_str());
+		return SQRESULT_ERROR;
+	}
+
+	g_saveFileManager->DeleteFileAsync<context>(dir / fileName);
+	return SQRESULT_NOTNULL;
+}
+
+// The param is not optional because that causes issues :)
+ADD_SQFUNC("array<string>", NS_InternalGetAllFiles, "string path", "", ScriptContext::CLIENT | ScriptContext::UI | ScriptContext::SERVER)
+{
+	// depth 1 because this should always get called from Northstar.Custom
+	Mod* mod = g_pSquirrel<context>->getcallingmod(sqvm, 1);
+	fs::path dir = savePath / fs::path(mod->m_ModDirectory).filename();
+	std::string pathStr = g_pSquirrel<context>->getstring(sqvm, 1);
+	fs::path path = dir;
+	if (pathStr != "")
+		path = dir / pathStr;
+	if (!IsPathSafe(pathStr, dir))
+	{
+		g_pSquirrel<context>->raiseerror(
+			sqvm,
+			fmt::format("File name invalid ({})! Make sure it results in a path inside your mod's save folder.", pathStr, mod->Name)
+				.c_str());
+		return SQRESULT_ERROR;
+	}
+	try
+	{
+		g_pSquirrel<context>->newarray(sqvm, 0);
+		for (const auto& entry : fs::directory_iterator(path))
+		{
+			g_pSquirrel<context>->pushstring(sqvm, entry.path().filename().string().c_str());
+			g_pSquirrel<context>->arrayappend(sqvm, -2);
+		}
+		return SQRESULT_NOTNULL;
+	}
+	catch (std::exception ex)
+	{
+		spdlog::error("DIR ITERATE FAILED! Is the path valid?");
+		g_pSquirrel<context>->raiseerror(sqvm, ex.what());
+		return SQRESULT_ERROR;
+	}
+}
+
+ADD_SQFUNC("bool", NSIsFolder, "string path", "", ScriptContext::CLIENT | ScriptContext::UI | ScriptContext::SERVER)
+{
+	Mod* mod = g_pSquirrel<context>->getcallingmod(sqvm);
+	fs::path dir = savePath / fs::path(mod->m_ModDirectory).filename();
+	std::string pathStr = g_pSquirrel<context>->getstring(sqvm, 1);
+	fs::path path = dir;
+	if (pathStr != "")
+		path = dir / pathStr;
+	if (!IsPathSafe(pathStr, dir))
+	{
+		g_pSquirrel<context>->raiseerror(
+			sqvm,
+			fmt::format("File name invalid ({})! Make sure it results in a path inside your mod's save folder.", pathStr, mod->Name)
```

Comment:
Another reason for this to be raised is the usage of non-ASCII characters, I think it should be mentioned here:
```suggestion
			fmt::format("File name invalid ({})! Make sure it does not contain any non-ASCII character, and results in a path inside your mod's save folder.", pathStr, mod->Name)
```